e2e/files.spec.ts

```diff
@@ -0,0 +1,283 @@
+import { waitForInitialPageLoad } from './helpers/utils'
+
+const openFileSharingSettings = async (page: Page) => {
+  await testAuth.step('First user can open File Sharing Settings', async () => {
+    await page.getByRole('button', { name: 'File Tools' }).click()
+    await page.getByRole('menuitem', { name: 'File Sharing Settings' }).click()
+    await expect(
+      page.getByRole('heading', { name: 'File Sharing Settings' }),
+    ).toBeVisible()
+  })
+}
+
+const expectNoAccessPage = async (page: Page) => {
+  await expect(
+    page.getByRole('heading', {
+      name: 'Sorry, no access to this page.',
+    }),
+  ).toBeVisible()
+  await expect(
+    page.getByText('You are not authorized to access the page requested.'),
+  ).toBeVisible()
+}
+
+const confirmSharingSettings = async (
+  firstUserPage: Page,
+  firstUserName: string,
+  secondUserName: string,
+  secondUserVisible: boolean,
+) => {
+  await testAuth.step(
+    'First user is listed in File Sharing settings',
+    async () => {
+      await expect(
+        firstUserPage.getByRole('cell', {
+          name: firstUserName,
+        }),
+      ).toBeVisible()
+    },
+  )
+
+  await testAuth.step(
+    `Second user is ${
+      secondUserVisible ? '' : 'not'
+    } listed in File Sharing settings`,
+    async () => {
+      const secondUserCell = firstUserPage.getByRole('cell', {
+        name: secondUserName,
+      })
+      if (secondUserVisible) {
+        await expect(secondUserCell).toBeVisible()
+      } else {
+        await expect(secondUserCell).not.toBeVisible()
+      }
+    },
+  )
+}
+
+const confirmAndClosePermissionsSavedAlert = async (page: Page) => {
+  const permissionsAlert = page.getByRole('alert').filter({
+    has: page.getByText('Permissions were successfully saved to Synapse'),
+  })
+  await expect(permissionsAlert).toBeVisible()
+  await permissionsAlert.getByRole('button').click()
+  await expect(permissionsAlert).not.toBeVisible()
+}
+
+const saveFileSharingSettings = async (page: Page) => {
+  const saveButton = page.getByRole('button', { name: 'Save' })
+  await saveButton.click()
+  await expect(saveButton).not.toBeVisible()
+
+  // Modal is closed after settings are saved
+  await expect(
+    page.getByRole('heading', { name: 'File Sharing Settings' }),
+  ).not.toBeVisible()
+
+  await confirmAndClosePermissionsSavedAlert(page)
+}
+
+let userProject: Project
+let fileHandleIds: string[] = []
+
+test.describe('Files', () => {
+  testAuth.beforeAll(async ({ browser, storageStatePaths }) => {
+    const userContext = await browser.newContext({
+      storageState: storageStatePaths['swc-e2e-user'],
+    })
+    const userPage = await userContext.newPage()
+    const userAccessToken = await getAccessTokenFromCookie(userPage)
+
+    // create project
+    const userProjectName = generateEntityName('project')
+    const userProjectId = await createProject(
+      userProjectName,
+      userAccessToken,
+      userPage,
+    )
```

Comment:
Create the Project with the test user's access token so that the project is initially only accessible by the test user